### PR TITLE
Fix for 404 spam when not logged in

### PR DESCRIPTION
**Pull request by chiguire**

### CODE REVIEW

File: frontend/BossAlienMediaPlayer/src/app/http.service.ts [hunk 1]
```diff
@@ -43,6 +43,10 @@ export class HttpService {
       err => 
       {
         this.requestInFlight = false;
+        if (err.status == 404) // This prevents the pesky error from /api/queue when not logged in
```
Comment (kevadsett) [inline frontend/BossAlienMediaPlayer/src/app/http.service.ts:46]: Hmm, I'm not sure blanket-ignoring all 404's is the best solution to this problem. Maybe better would be to only send the requests if we know we are logged in?

**Review by kevadsett — CHANGES_REQUESTED** [review]

File: frontend/BossAlienMediaPlayer/src/app/http.service.ts (hunk 1 above)
Comment (chiguire) [reply frontend/BossAlienMediaPlayer/src/app/http.service.ts:46]: Ok, I have a better understanding on how to develop with Angular after doing the listeners thing, I'll double check the queue component, which I think is the culprit for this

Comment (chiguire) [reply frontend/BossAlienMediaPlayer/src/app/http.service.ts:46]: Still, retrying a 404 makes no sense, so I think it should stay

Comment (chiguire) [reply frontend/BossAlienMediaPlayer/src/app/http.service.ts:44]: I've located where the extra call was done, I'll still keep this one for the above reason

File: frontend/BossAlienMediaPlayer/src/app/http.service.ts [hunk 2]
```diff
@@ -26,13 +26,11 @@ export class HttpService {
     this.lastRequestTime = Date.now();
     this.retryDelay = retryDelay;
     let http$ = this.http.get<T>(url, {observe: "response"});
-    http$.pipe(
```
Comment (chiguire) [inline frontend/BossAlienMediaPlayer/src/app/http.service.ts:29]: @kevadsett I've removed this bit and moved it after the results arrive. I checked with print debugging and this pipe was never executed, so the version was never checked. Once moved inside the subscribe it worked as intended.

**Pull request merged**